src/Symfony/Bundle/SecurityBundle/DependencyInjection/Security/Factory/JsonLoginFactory.php

```diff
@@ -24,6 +24,8 @@
  */
 class JsonLoginFactory extends AbstractFactory implements AuthenticatorFactoryInterface
 {
+    public const PRIORITY = -40;
```

Comment:
Could we invert the priority of FormLoginFactory with this one to address https://github.com/symfony/symfony/issues/41959?

Reply:
Wouldn't that create the opposite? (by-passing the form login authenticator when submitting the form) I think adding the content type check as suggested in the issue is a better fix.

Reply:
In which case would it bypass the form login one? JsonLoginAuthenticator already has a content-type check in `supports()` (looks for json) so making it run first should be enough, but I'm fine either way

Reply:
Adding a content type check seems like it's more descriptive than changing a priority.  My concern would be that the priority might get changed later for another good reason and re-introduce the problem it was changed to fix.

Reply:
Fix implemented here:  https://github.com/symfony/symfony/pull/41993